Refactor shopify-docker repository

aaron-medina-aligent opened the pull request:
Created Docker base image to Aligent’s Dockerhub
Updated pipeline and dockerfile to use base image
Created Aliases for docker commands
Moved a few lines of dockerfile commands to entrypoint.sh Changed branch from main to production
Restructured shopify theme folder handling
Wrote script for downloading the theme from client repo
Updated README file
Changed production deployment to trigger: manual

Commit:
Refactor shopify-docker repository

Created Docker base image to Aligent’s Dockerhub
Updated pipeline and dockerfile to use base image
Created Aliases for docker commands
Moved a few lines of dockerfile commands to entrypoint.sh
Changed branch from main to production
Restructured shopify theme folder handling
Wrote script for downloading the theme from client repo

daniel-vdp commented on README.md:
````diff
@@ -0,0 +1,163 @@
+
+Navigate to the repository and run the following command so it will not include the repository meta-data and README.md.
+
+```bash
````
I thought we were going to publish this to docker hub? :thinking: Then the devs can just use a bash alias to execute

aaron-medina-aligent replied:
The base image has been published to Docker hub. But as for the local dev env of the devs, they need a little bit more of customisation for the wrappers and file permissions.

daniel-vdp replied:
What's stopping us from just running the docker container with an alias without cloning everything down? The extra customisation can be built into the container

gowrizrh commented on docker-compose.yml:
```diff
@@ -0,0 +1,21 @@
+services:
```
let's rename this file to `compose.yaml` to align with rest of the company

aaron-medina-aligent replied:
Done

daniel-vdp commented on assets/entrypoint.sh:
```diff
@@ -0,0 +1,38 @@
+su shopifyuser
+
+# Execute the passed command or default to bash
+exec "$@"
```
Instead of just executing the passed command we can assume the user is passing an arg for the shopify cli tool (or the wrapper in this case). Then if I have `alias shopify="docker run ..."` I can just type `shopify build`. 

With the current method I would need to type `shopify shopify-cli build`

aaron-medina-aligent replied:
The dev would just need to run:
`shopify env int ` - for building and running the container in detached mode
`shopify env` stop/start - for stopping and starting the container

[README.md](https://github.com/aligent/shopify-docker/blob/feature/DO-1709-Refactor-shopify-docker-repo/README.md)

gowrizrh commented on set-up.sh:
```diff
@@ -0,0 +1,145 @@
+)
+
+# Detect the shell and determine the appropriate profile file
+if [ -n "$BASH_VERSION" ]; then
```
not really sure that we would want to modify shell profiles, but keen to know others thoughts on this

daniel-vdp replied:
Yeah agree with this. I'm not a fan of modifying a user's profile with a script. I think best to provide what the user should add and get them to do it themselves

aaron-medina-aligent replied:
Removed in the script, just added a guide in README

gowrizrh commented on set-up.sh:
```diff
@@ -0,0 +1,145 @@
+
+# Define the function for 'shopify' to support 'env' commands
+FUNCTION_COMMANDS=$(cat << 'EOF'
+shopify() {
```
ideally not having custom scripts or aliases would play well with manta. if we just have plain docker compose commands, we can orchestrate that with manta. manta would just call `docker compose start` and `docker compose stop` on `manta shopify X`

aaron-medina-aligent replied:
Got it, will consider this in the next major revamp of this repo.

daniel-vdp commented on bitbucket-pipelines.yml:
```diff
@@ -0,0 +1,53 @@
+        script:
+          - echo "Starting THEME CHECK..."
+          - ruby -v && node -v && npm -v && shopify version
+          - find . -exec ls -ld --time-style=long-iso {} \; | awk '{print $6, $7, $8, $9}'
```
This line probably needs a comment, I'm not sure what it's doing or why :sweat_smile:

aaron-medina-aligent replied:
Updated

daniel-vdp commented on bitbucket-pipelines.yml:
```diff
@@ -0,0 +1,53 @@
+
+pipelines:
+  branches:
+    feature/*:
```
Is this like a linting step? We'd generally do that on any pull request

aaron-medina-aligent replied:
Updated

daniel-vdp commented on bitbucket-pipelines.yml:
```diff
@@ -0,0 +1,53 @@
+          deployment: Production
+
+  pull-requests:
+    staging:
```
I don't think these two are required. Might be best to remove these and move the theme-check here instead

daniel-vdp reviewed (changes requested):
Couple questions

Commit:
Renamed docker-compose.yml to compose.yaml
Removed adding of aliases via script
Modified listing of theme files in pipeline
Added linting to PR in bitbucket pipeline
Updated README

aaron-medina-aligent reviewed (commented):
I have updated the requested changes

Commit:
changed 2nd Step to git clone
